lib/livebook_cli/deploy.ex

```diff
@@ -0,0 +1,208 @@
+defmodule LivebookCLI.Deploy do
+  import LivebookCLI.Utils
+  alias Livebook.Teams
+
+  @behaviour LivebookCLI.Task
+
+  @deploy_key_prefix Teams.Constants.deploy_key_prefix()
+  @teams_key_prefix Teams.Constants.teams_key_prefix()
+
+  @impl true
+  def usage() do
+    """
+    Usage: livebook deploy [options] filename|directory
+
+    ## Available options
+
+      --deploy-key        Sets the deploy key to authenticate with Livebook Teams
+      --teams-key         Sets the Teams key to authenticate with Livebook Teams and encrypt the Livebook app
+      --deployment-group  The deployment group name which you want to deploy to
+
+    The --help option can be given to print this notice.
+
+    ## Examples
+
+    Deploys a single notebook:
+
+        livebook deploy --deploy-key="lb_dk_..." --teams-key="lb_tk_..." --deployment-group "online" path/to/app1.livemd
+
+    Deploys multiple notebooks:
+
+        livebook deploy --deploy-key="lb_dk_..." --teams-key="lb_tk_..." --deployment-group "online" path/to/*.livemd\
+    """
+  end
+
+  @switches [
+    deploy_key: :string,
+    teams_key: :string,
+    deployment_group: :string
+  ]
+
+  @impl true
+  def call(args) do
+    Application.put_env(:livebook, :persist_storage, false)
+    {:ok, _} = Application.ensure_all_started(:livebook)
+    config = config_from_args(args)
+    ensure_config!(config)
+
+    team = authenticate_cli!(config)
+    deploy_to_teams(team, config)
+  end
+
+  defp config_from_args(args) do
+    {opts, path} = OptionParser.parse!(args, strict: @switches)
+
+    %{
+      path: List.flatten(path),
```

Comment:
I believe it is always a list, no need for flattening. I would also have `paths` as plural. :)

```suggestion
    {opts, paths} = OptionParser.parse!(args, strict: @switches)

    %{
      paths: paths,
```